src/main/scala/rocket/RocketCore.scala

```diff
@@ -572,13 +639,17 @@ class Rocket(tile: RocketTile)(implicit p: Parameters) extends CoreModule()(p)
     mem_reg_hls_or_dv := io.dmem.req.bits.dv
     mem_reg_pc := ex_reg_pc
     // IDecode ensured they are 1H
-    mem_reg_wdata := alu.io.out
+    mem_reg_wdata := Mux(ex_reg_set_vconfig, ex_new_vl.getOrElse(alu.io.out), alu.io.out)
     mem_br_taken := alu.io.cmp_out
 
+
     when (ex_ctrl.rxs2 && (ex_ctrl.mem || ex_ctrl.rocc || ex_sfence)) {
       val size = Mux(ex_ctrl.rocc, log2Ceil(xLen/8).U, ex_reg_mem_size)
       mem_reg_rs2 := new StoreGen(size, 0.U, ex_rs(1), coreDataBytes).data
     }
+    if (usingVector) { when (ex_reg_set_vconfig) {
+      mem_reg_rs2 := ex_new_vconfig.get.asUInt
+    } }
```

Comment:
Use it for trap?

Reply:
What do you mean here?

Reply:
Oh, small notes when doing code review. I understand it now.